### PR TITLE
refactor(components): [pagination] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by metanas**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by JeremyWuuuuu** [comment]
Hi @metanas , thank you for contributing. Would you mind following the example [here](https://github.com/element-plus/element-plus/issues/6114#issuecomment-1125552539) for me to process it before getting started?

Just want to make sure that your PR is not duplicated. Thanks. PRs made without mentioning in the issue will not be accepted. 🙏

**Comment by metanas** [comment]
> Hi @metanas , thank you for contributing. Would you mind following the example [here](https://github.com/element-plus/element-plus/issues/6114#issuecomment-1125552539) for me to process it before getting started?
> 
> Just want to make sure that your PR is not duplicated. Thanks. PRs made without mentioning in the issue will not be accepted. pray

Thank you @JeremyWuuuuu and Sorry for that miss, I will remember it for my next PR.

File: packages/components/pagination/src/components/next.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  pageCount: {
+    type: Number,
+    default: 50,
+  },
+  nextText: {
+    type: String,
+    default: '',
+  },
+})
```
Comment (btea) [inline packages/components/pagination/src/components/next.ts:19]:
```suggestion
} as const)
```

File: packages/components/pagination/src/components/pager.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  pageCount: {
+    type: Number,
+    required: true,
+  },
+  pagerCount: {
+    type: Number,
+    default: 7,
+  },
+  disabled: Boolean,
+})
```
Comment (btea) [inline packages/components/pagination/src/components/pager.ts:19]:
```suggestion
} as const)
```

File: packages/components/pagination/src/components/prev.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  disabled: Boolean,
+  currentPage: {
+    type: Number,
+    default: 1,
+  },
+  prevText: {
+    type: String,
+    default: '',
+  },
+})
```
Comment (btea) [inline packages/components/pagination/src/components/prev.ts:15]:
```suggestion
} as const)
```

File: packages/components/pagination/src/components/size.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  popperClass: {
+    type: String,
+    default: '',
+  },
+  disabled: Boolean,
+  size: {
+    type: String,
+    default: 'default',
+  },
+})
```
Comment (btea) [inline packages/components/pagination/src/components/size.ts:23]:
```suggestion
} as const)
```

File: packages/components/pagination/src/components/total.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { buildProps } from '@element-plus/utils'
+import type Total from './total.vue'
+import type { ExtractPropTypes } from 'vue'
+
+export const paginationTotalProps = buildProps({
+  total: {
+    type: Number,
+    default: 1000,
+  },
+})
```
Comment (btea) [inline packages/components/pagination/src/components/total.ts:10]:
```suggestion
} as const)
```

**Comment by JeremyWuuuuu** [comment]
@metanas This PR is pretty huge for people to review. Would you mind splitting your commits or splitting them into small PRs for easier reviewing? 

For example:

```
commits: next
commits: prev
commits: pager
commits: pagination

```

**Review by sxzz — COMMENTED** [review]
Please remove it from `internal/build/src/type-unsafe.json`

**Pull request closed without merging**